src/dqe_idx.erl

```diff
@@ -227,6 +228,28 @@ metrics(Collection) ->
 
 %%--------------------------------------------------------------------
 %% @doc
+%% Returns a list of metric path suffixes of `Depth' that are prefixed
+%% by the given probe `Prefix', which can also be empty.
+%%
+%% For example:
+%% metrics(<<"collection">>, <<>>, 1) -> [<<"base">>].
+%% metrics(<<"collection">>, <<"base">>, 1) -> [<<"cpu">>].
+%% metrics(<<"collection">>, <<>>, 2) -> [[<<"base">>.<<"cpu">>]].
+%%
+%% Actual results are binary-encoded in `dproto' encoding.
+%% @end
+%%--------------------------------------------------------------------
+-spec metrics(Collection::collection(),
+              Prefix::metric(),
```

Comment:
This type spec is incorrect as you have pointed out.  I noticed that the namespaces/2 function, accepts `metric` arguments as either lists or binaries encoded using `dproto:metric_from_list`.  See for example:https://github.com/dalmatinerdb/dqe_idx_pg/blob/master/src/dqe_idx_pg.erl#L119-L134
I believe that this is ok, but leads to inconsistencies as per the above.  The right thing to do is to allow the same convention for any metric() argument.  If that is the case, we could correct the type specs of callback inputs to something like [this](https://github.com/dalmatinerdb/dqe_idx_pg/blob/master/src/dqe_idx_pg.erl#L245) whenever a metric is involved.  I am happy to do this if you agree that this is a good course of action to eliminate the confusion.